ledger/src/blockstore_meta.rs

```diff
@@ -374,9 +374,41 @@ pub struct AddressSignatureMeta {
     pub writeable: bool,
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq, Eq)]
-pub struct PerfSample {
+/// Performance information on the validator execution during a time slice.
+///
+/// Older versions should only arise as a result of deserialization of entries stored by a previous
+/// version of the validator.  Current version should only produce [`PerfSampleV2`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PerfSample {
+    V1(PerfSampleV1),
+    V2(PerfSampleV2),
+}
+
+impl From<PerfSampleV1> for PerfSample {
+    fn from(value: PerfSampleV1) -> PerfSample {
+        PerfSample::V1(value)
+    }
+}
+
+impl From<PerfSampleV2> for PerfSample {
+    fn from(value: PerfSampleV2) -> PerfSample {
+        PerfSample::V2(value)
+    }
+}
+
+/// Older version of [`PerfSample`].
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct PerfSampleV1 {
+    pub num_transactions: u64,
+    pub num_slots: u64,
+    pub sample_period_secs: u16,
+}
+
+/// Current version of the [`PerfSample`].
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct PerfSampleV2 {
     pub num_transactions: u64,
+    pub num_non_vote_transactions: u64,
```

Comment:
Trying to be as friendly to future us as possible, I'm thinking about the following sequence which does come up:
1) Someone is running master or some branch that has this commit (ie inserting V2s)
2) They run into an issue unrelated to this change that causes them to want to bisect (with the same ledger, maybe via `solana-ledger-tool verify`)
3) This person at some point ends up trying to run a commit older than the one this PR will introduce
4) The older commit isn't aware of PerfSampleV1 & V2, only the original PerfSample struct which is 18 bytes (as is V1) and as such, will try to deserialize blockstore entries into the original struct

Given that we use the serialize / deserialize functions (and not the option), deserializing an instance of V2 (26 bytes) that was inserted in 1) above (ie with this commit) will not error in 4) as an original PerfSample, since the deserialize function [allows trailing bytes by default](https://docs.rs/bincode/1.3.3/bincode/config/index.html#options-struct-vs-bincode-functions) (original has one less u64 so 8 trailing bytes).

Even though we won't get an error, the ordering will have changed so we'll be interpreting fields incorrectly. This can be avoided if we move the new field to the end of the struct. I realize this is pretty brittle, but it is better than doing nothing. All this being said, I struggle to see a scenario where we'd be debugging something of the sort and examining this column, but better to have one less possible foot-gun I guess.

Note: I'm pretty sure bincode serializes struct fields in the order specified, even if the compiler would want to rearrange things for better packing (ie multiple u16's in the same struct as a u64).